### PR TITLE
Feature/firebase admin module basic implementation

### PR DESCRIPTION
**Pull request by adrianmjim**

### CODE REVIEW

File: packages/nestjs-firebase-admin/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (notaphplover) [inline packages/nestjs-firebase-admin/lib/index.ts]: we should export `FirebaseAdminModule` and `FirebaseProviderType`

File: packages/nestjs-firebase-admin/lib/modules/FirebaseAdminModule.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { getMachineLearning, MachineLearning } from 'firebase-admin/machine-learning';
+import { getMessaging, Messaging } from 'firebase-admin/messaging';
+import { getProjectManagement, ProjectManagement } from 'firebase-admin/project-management';
+import { getRemoteConfig, RemoteConfig } from 'firebase-admin/remote-config';
+import { getSecurityRules, SecurityRules } from 'firebase-admin/security-rules';
+import { getStorage, Storage } from 'firebase-admin/storage';
+
+import { AppAsyncOptions } from './AppAsyncOptions';
+
+type FirebaseProviderType =
```
Comment (notaphplover) [inline packages/nestjs-firebase-admin/lib/modules/FirebaseAdminModule.ts]: This type is being used at `FirebaseAdminModule.forRoot`, so it should be exposed in order to expose it at `index` module

**Review by notaphplover — COMMENTED** [review]
Nice, looks great. We can move enums to different files.

Should we provide decorators in order to allow injecting dependencies based on interfaces such as firebase-admin `Database`?

File: packages/nestjs-firebase-admin/lib/index.ts (hunk 1 above)
Comment (adrianmjim) [reply packages/nestjs-firebase-admin/lib/index.ts]: Sorry, but I don't know what We have to export in a library context. Sorry for ignorance. I'm adding it!

File: packages/nestjs-firebase-admin/lib/modules/FirebaseAdminModule.ts (hunk 2 above)
Comment (adrianmjim) [reply packages/nestjs-firebase-admin/lib/modules/FirebaseAdminModule.ts]: Same https://github.com/cuaklabs/nestjs-meow/pull/1#discussion_r755593904

**Comment by adrianmjim** [comment]
> Nice, looks great. We can move enums to different files.
> 
> Should we provide decorators in order to allow injecting dependencies based on interfaces such as firebase-admin `Database`?

Great!

Do we have enums? Do you mean types?

I think provide decorators could be a great approach if We can not provide direct injection without decorators. But this could create a inconsistency problem in API library

File: packages/nestjs-firebase-admin/lib/index.ts (hunk 1 above)
Comment (notaphplover) [reply packages/nestjs-firebase-admin/lib/index.ts]: It's ok :), we're just creating a module used as the library's entrypoint. Every type required to use the library should be added here

**Comment by notaphplover** [comment]
> > Nice, looks great. We can move enums to different files.
> > Should we provide decorators in order to allow injecting dependencies based on interfaces such as firebase-admin `Database`?
> 
> Great!
> 
> Do we have enums? Do you mean types?

Yeah, I mean types :)

> I think provide decorators could be a great approach if We can not provide direct injection without decorators. But this could create a inconsistency problem in API library

could you provide an example of inconsistency?

**Comment by adrianmjim** [comment]
> > > Nice, looks great. We can move enums to different files.
> > > Should we provide decorators in order to allow injecting dependencies based on interfaces such as firebase-admin `Database`?
> > 
> > 
> > Great!
> > Do we have enums? Do you mean types?
> 
> Yeah, I mean types :)
> 
> > I think provide decorators could be a great approach if We can not provide direct injection without decorators. But this could create a inconsistency problem in API library
> 
> could you provide an example of inconsistency?

@notaphplover Check moved types

Of course! Basic approach of injection:
```ts
export class ServiceExample {
  public constructor(private readonly auth: Auth) {}
}
```

Injection with DatabaseDecorator:  
```ts
export class ServiceExample {
  public constructor(@InjectDatabase() private readonly database: Database) {}
}
```

If We use decorators, We have 2 approachs to inject services of firebase. I think this could create a inconsistency.

**Comment by notaphplover** [comment]
> If We use decorators, We have 2 approachs to inject services of firebase. I think this could create a inconsistency.

Maybe, but it's the only way I think users can easily inject everything. Docs could explain this as two different approaches, users can choose whatever they want.

**Review by notaphplover — APPROVED** [review]

**Pull request merged**